Added the ability to pass arbitrary headers.

omadawn opened the pull request:
Added a --header | -H option to allow the passing of arbitrary headers.

Fixes #186

omadawn committed:
Added the ability to pass additional headers.

Updated documentation.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/terraform-aws-consul?pullRequest=187)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/terraform-aws-consul?pullRequest=187)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)


---
**Forster, Aaron** seems not to be a GitHub user.
You need a GitHub account to be able to sign the CLA. If you already have a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/terraform-aws-consul?pullRequest=187).</sub>

brikis98 commented on modules/install-consul/install-consul:
```diff
@@ -164,14 +168,26 @@ function create_consul_install_paths {
+
+    for value in "${addtl_headers[@]}"
+    do
+      headers+="-H \"$value\" "
```
Slightly cleaner to make `headers` an array (declare it up above `local -a headers=()`), appending to it here via `headers+=("-H \"$value\"")`, and expanding it in the `curl` command below with `${headers[@]}`.

brikis98 reviewed (commented):
Thanks for the PR!

omadawn commented on modules/install-consul/install-consul:
```diff
@@ -25,13 +27,15 @@ function print_usage {
-  echo -e "  --ca-file-path\t\tPath to a PEM-encoded certificate authority used to encrypt and verify authenticity of client and server connections. Will be installed under <install-path>/tls/ca."
-  echo -e "  --cert-file-path\t\tPath to a PEM-encoded certificate, which will be provided to clients or servers to verify the agent's authenticity. Will be installed under <install-path>/tls. Must be provided along with --key-file-path."
-  echo -e "  --key-file-path\t\tPath to a PEM-encoded private key, used with the certificate to verify the agent's authenticity. Will be installed under <install-path>/tls. Must be provided along with --cert-file-path"
+  echo -e "  --help | -h\t\tPrints this usage message."
```
I'll try and get to that this weekend.

omadawn committed:
Update modules/install-consul/install-consul

Co-authored-by: Yevgeniy Brikman <brikis98@users.noreply.github.com>

omadawn committed:
Update modules/install-consul/install-consul

According to google arrays are always local so that bit is redundant but I like it for consistency.

Co-authored-by: Yevgeniy Brikman <brikis98@users.noreply.github.com>

omadawn committed:
Update modules/install-consul/README.md

Meant to add that, ty.

Co-authored-by: Yevgeniy Brikman <brikis98@users.noreply.github.com>

omadawn commented on modules/install-consul/README.md:
````diff
@@ -44,20 +44,25 @@ example](https://github.com/hashicorp/terraform-aws-consul/tree/master/examples/
 ```
 
+```
+install-consul --download-url https://staticassets.example.com/consul/1.2.3/consul_1.2.3_linux_amd64.zip -H "X-API-Key: 123fks9213432"
````
We have no idea what customers would use for storing static assets. My use case actually requires the API key in the header `X-JFrog-Art-Api` but that would be different for nexus or some generic FTP server.
On top of that a generic header is more flexible so we wouldn't need another PR when someone wants to add a header for anything else.

If the header was standard across platforms we could add both but I didn't want to add any specifics for a single unrelated technology.

brikis98 replied:
Got it. I was wondering if HashiCorp/Consul itself had some native header it was requiring now (e.g., for Consul Enterprise), but if that's not the case, sticking with a generic header makes sense.

omadawn committed:
Update modules/install-consul/install-consul

Co-authored-by: Yevgeniy Brikman <brikis98@users.noreply.github.com>

omadawn committed:
Update modules/install-consul/install-consul

Co-authored-by: Yevgeniy Brikman <brikis98@users.noreply.github.com>
